packages/core/src/textures/resources/ImageBitmapResource.ts

```diff
@@ -1,27 +1,196 @@
+                });
+
+                if (this.destroyed) return;
+
+                const imageBlob = await response.blob();
+
+                if (this.destroyed) return;
+
+                const imageBitmap = await createImageBitmap(imageBlob, {
+                    premultiplyAlpha: this.alphaMode === null || this.alphaMode === ALPHA_MODES.UNPACK
+                        ? 'premultiply' : 'none',
+                });
+
+                if (this.destroyed) return;
+
+                this.source = imageBitmap;
+                this.update();
+
+                resolve(this);
+            }
+            catch (e)
+            {
+                if (this.destroyed) return;
+
+                reject(e);
+                this.onError.emit(e);
+            }
+        });
+
+        return this._load;
+    }
+
     /**
-     * @param source - Image element to use
+     * Upload the image bitmap resource to GPU.
+     * @param renderer - Renderer to upload to
+     * @param baseTexture - BaseTexture for this resource
+     * @param glTexture - GLTexture to use
+     * @returns {boolean} true is success
      */
-    // eslint-disable-next-line @typescript-eslint/no-useless-constructor
-    constructor(source: ImageBitmap)
+    override upload(renderer: Renderer, baseTexture: BaseTexture, glTexture: GLTexture): boolean
     {
-        super(source);
+        if (!(this.source instanceof ImageBitmap))
+        {
+            this.load();
+
+            return false;
+        }
+
+        if (typeof this.alphaMode === 'number')
+        {
+            baseTexture.alphaMode = this.alphaMode;
+        }
+
+        return super.upload(renderer, baseTexture, glTexture);
+    }
+
+    /** Destroys this resource. */
+    override dispose(): void
+    {
+        if (this.source instanceof ImageBitmap)
+        {
+            this.source.close();
+        }
+
+        super.dispose();
+
+        this._load = null;
     }
 
     /**
      * Used to auto-detect the type of resource.
      * @param {*} source - The source object
-     * @returns {boolean} `true` if source is an ImageBitmap
+     * @returns {boolean} `true` if current environment support ImageBitmap, and source is string or ImageBitmap
      */
-    static test(source: unknown): source is ImageBitmap
+    static override test(source: unknown): source is string | ImageBitmap
     {
-        return !!globalThis.createImageBitmap && typeof ImageBitmap !== 'undefined' && source instanceof ImageBitmap;
+        return !!globalThis.createImageBitmap && typeof ImageBitmap !== 'undefined'
+            && (typeof source === 'string' || source instanceof ImageBitmap);
+    }
+
+    private static __DUMMY: ICanvas;
+
+    private static get _DUMMY(): ICanvas
+    {
+        if (ImageBitmapResource.__DUMMY) return ImageBitmapResource.__DUMMY;
+
+        ImageBitmapResource.__DUMMY = settings.ADAPTER.createCanvas(0, 0);
+
+        return ImageBitmapResource.__DUMMY;
     }
```

Comment:
"Dummy" naming could be improved :) Also comments are helpful. 

```suggestion
    /** Temporary empty canvas to use when loading URLs */
    private static _EMPTY: ICanvas;
   
    /** Get cached temp canvas */
    private static get EMPTY(): ICanvas
    {
        ImageBitmapResource._EMPTY = ImageBitmapResource._EMPTY ?? settings.ADAPTER.createCanvas(0, 0);

        return ImageBitmapResource._EMPTY;
    }
```

Reply:
Done in ca8527757166676cf4d8e4482d072bd513469093.